ctest.h

```diff
@@ -449,9 +491,42 @@ static void sighandler(int signum)
 }
 #endif
 
-int ctest_main(int argc, const char *argv[]);
+#ifdef CTEST_IMPL_CLANG_POISIONED_GLOBALS
+#include <sanitizer/asan_interface.h>
+#endif
+
+static struct ctest* ctest_impl_adjacent(const struct ctest* test,
+                                         int direction, size_t offset)
+{
+    const volatile unsigned int* magic = (const unsigned int*)
+                                         ((const char*)
+                                          (test + direction) + offset);
+
+#if CTEST_IMPL_POISIONED_GLOBALS
+    while (ASAN_UNPOISON_MEMORY_REGION(magic, sizeof(*magic)), *magic == 0) {
```

Comment:
An alternative to trying to work around sanitizers would be to change the test discovery method to not involve the use of a linker set. Linker sets are a handy tool, but are difficult to do portably (as this PR emphasizes). An approach that shouldn't anger sanitizers would be to construct a global linked list of tests using static constructors (mentioned as a solution to a different issue in https://github.com/bvdberg/ctest/pull/35#issuecomment-442108765). I think we should consider this approach since it may kill two birds with one stone (fixes both the sanitizer issue and the Windows incremental linking issue while adding comparatively little platform-specific code).